Content.Server/VendingMachines/VendingMachineComponent.cs

```diff
@@ -1,284 +1,58 @@
 using System;
-using System.Collections.Generic;
-using System.Linq;
-using Content.Server.Popups;
-using Content.Server.Power.Components;
 using Content.Server.UserInterface;
 using Content.Server.WireHacking;
-using Content.Shared.Access.Components;
-using Content.Shared.Access.Systems;
-using Content.Shared.Acts;
-using Content.Shared.Interaction;
 using Content.Shared.Sound;
 using Content.Shared.VendingMachines;
 using Robust.Server.GameObjects;
-using Robust.Shared.Audio;
 using Robust.Shared.GameObjects;
-using Robust.Shared.IoC;
-using Robust.Shared.Localization;
-using Robust.Shared.Player;
-using Robust.Shared.Prototypes;
-using Robust.Shared.Random;
 using Robust.Shared.Serialization.Manager.Attributes;
 using Robust.Shared.ViewVariables;
+using Content.Server.VendingMachines.systems;
 using static Content.Shared.Wires.SharedWiresComponent;
 
 namespace Content.Server.VendingMachines
 {
     [RegisterComponent]
-    [ComponentReference(typeof(IActivate))]
-    public class VendingMachineComponent : SharedVendingMachineComponent, IActivate, IBreakAct, IWires
+    public class VendingMachineComponent : SharedVendingMachineComponent, IWires
     {
-        [Dependency] private readonly IEntityManager _entMan = default!;
-        [Dependency] private readonly IRobustRandom _random = default!;
-        [Dependency] private readonly IPrototypeManager _prototypeManager = default!;
-
-        private bool _ejecting;
-        private TimeSpan _animationDuration = TimeSpan.Zero;
+        public bool Ejecting;
+        public TimeSpan AnimationDuration = TimeSpan.Zero;
         [DataField("pack")]
-        private string _packPrototypeId = string.Empty;
-        private string _spriteName = "";
-
-        private bool Powered => !_entMan.TryGetComponent(Owner, out ApcPowerReceiverComponent? receiver) || receiver.Powered;
-        private bool _broken;
-
+        public string PackPrototypeId = string.Empty;
+        public string SpriteName = "";
+        public bool Broken;
+        [DataField("speedLimiter")]
+        public bool SpeedLimiter = true;
```

Comment:
Please add doc summary explaining what it's for, e.g. "Allows vending machine to throw items when true" or something. I'd also maybe call it "CanShoot" or something as it's more descriptive for code reasons?

Reply:
Thanks for the patience and help, made the changes